src/foam/lib/parse/Literal.java

```diff
@@ -31,6 +31,7 @@ public Literal(String s, Object v) {
   }
 
   public PStream parse(PStream ps, ParserContext x) {
+    if ( string ==null) return null;
```

Comment:
You should have a space between "==" and "null" as well as between "null" and ")" to match the FOAM style guide.